devito/mpi/distributed.py

```diff
@@ -190,6 +191,10 @@ def __init__(self, shape, dimensions, input_comm=None, topology=None):
                 global init_by_devito
                 init_by_devito = True
 
+                # If MPI0 logging is selected, only emit from rank 0
```

Comment:
I don't think this is a safe way to do it. You also end up here quite often when devito is called from an external program that initialized MPI already and that would completely turn off the logging on all ranks including 0.

Reply:
I second Mathias. Here you're potentially altering the overarching application behavior.

Also you probably don't need to add a new log level... just restrict logging to rank 0 (if MPI active) inside op._emit_perf_profile or whatever is called. No?